source/extensions/filters/http/aws_lambda/config.cc

```diff
@@ -194,11 +194,14 @@ void AWSLambdaConfigImpl::AWSLambdaStsRefresher::init(Event::Dispatcher &dispatc
     } else {
         ENVOY_LOG(debug, "{}: STS enabled without time based refresh",__func__);
     }
-    file_watcher_->addWatch(
+    // the result of this function MUST NOT be ignored (see [[nodiscard]]) - but
+    // what should we do with it?? TODO
+    absl::Status result = file_watcher_->addWatch(
         parent_->token_file_, Filesystem::Watcher::Events::Modified,
         [shared_this](uint32_t) {
           // Force timer callback to happen immediately to pick up the change.
           shared_this->timer_->enableTimer(std::chrono::milliseconds::zero());
+          return absl::OkStatus();
         });
```

Comment:
```suggestion
    RETURN_IF_NOT_OK(file_watcher_->addWatch(
        parent_->token_file_, Filesystem::Watcher::Events::Modified,
        [shared_this](uint32_t) {
          // Force timer callback to happen immediately to pick up the change.
          shared_this->timer_->enableTimer(std::chrono::milliseconds::zero());
          return absl::OkStatus();
        }));
```
In https://github.com/envoyproxy/envoy/commit/d3c90ed4a92119fa20167e61750efafa687ae2e5 they did similar things to handle similar filesystem things.
Given that the file_watcher used to never be able to return anything we would return after the function anyways so it seems the most like for like.